notebooks/outetts-text-to-speech/outetts-text-to-speech.ipynb

```diff
@@ -419,35 +341,294 @@
+    "\n",
+    "to_quantize = quantization_widget()\n",
+    "\n",
+    "to_quantize"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Fetch `skip_kernel_extension` module\n",
+    "import requests\n",
+    "\n",
+    "r = requests.get(\n",
+    "    url=\"https://raw.githubusercontent.com/openvinotoolkit/openvino_notebooks/latest/utils/skip_kernel_extension.py\",\n",
+    ")\n",
+    "open(\"skip_kernel_extension.py\", \"w\").write(r.text)\n",
+    "\n",
+    "%load_ext skip_kernel_extension"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare calibration dataset\n",
+    "[back to top ⬆️](#Table-of-contents:)\n",
+    "\n",
+    "The first step is to prepare calibration datasets for quantization. We will utilize [Filtered LibriTTS-R](https://huggingface.co/datasets/parler-tts/libritts_r_filtered) dataset as it was used to train the original model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%skip not $to_quantize.value\n",
+    "\n",
+    "from datasets import load_dataset\n",
+    "\n",
+    "libritts = load_dataset(\"parler-tts/libritts_r_filtered\", \"clean\", split=\"test.clean\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Quantize model\n",
+    "[back to top ⬆️](#Table-of-contents:)\n",
+    "\n",
+    "Below we run the quantize function which calls `nncf.quantize` on the OpenVINO IR model and collected dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%skip not $to_quantize.value\n",
+    "\n",
+    "import nncf\n",
+    "from functools import partial\n",
+    "import numpy as np\n",
+    "\n",
+    "\n",
+    "def transform_fn(item, interface):\n",
+    "    text_normalized = item[\"text_normalized\"]\n",
+    "    prompt = interface.prompt_processor.get_completion_prompt(text_normalized, interface.language, None)\n",
+    "    encoded = interface.prompt_processor.tokenizer(prompt, return_tensors=\"np\")\n",
+    "\n",
+    "    input_ids = encoded[\"input_ids\"]\n",
+    "    attention_mask = encoded[\"attention_mask\"]\n",
+    "    inputs = {\"input_ids\": input_ids, \"attention_mask\": attention_mask}\n",
+    "\n",
+    "    position_ids = np.cumsum(attention_mask, axis=1) - 1\n",
+    "    position_ids[attention_mask == 0] = 1\n",
+    "    inputs[\"position_ids\"] = position_ids\n",
+    "\n",
+    "    batch_size = input_ids.shape[0]\n",
+    "    inputs[\"beam_idx\"] = np.arange(batch_size, dtype=int)\n",
+    "\n",
+    "    return inputs\n",
+    "\n",
+    "hf_model = OVHFModel(model_dir, device.value).model\n",
+    "dataset = nncf.Dataset(libritts, partial(transform_fn, interface=interface))\n",
+    "\n",
+    "quantized_model = nncf.quantize(\n",
```

Comment:
I would suggest to use INT4 weight compression with dynamic quantization (A8W4). @KodiaqQ claim that the performance of such model is equal to the performance of the quantized model, but compression reate is higher for A8W4 model.

cc' @MaximProshin

Reply:
Please share numbers for both cases. If int4 is better, I'm ok to use that method.

Reply:
We discussed it offline and agreed to keep int8. At the same time the issue with SDPA will be analyzed in #16177. If resolved, we will update the notebook afterwards.